crates/sui-core/src/authority/consensus_tx_status_cache.rs

```diff
@@ -62,21 +76,34 @@ impl ConsensusTxStatusCache {
     }
 
     pub fn set_transaction_status(&self, pos: ConsensusPosition, status: ConsensusTxStatus) {
-        let mut inner = self.inner.write();
         if let Some(last_committed_leader_round) = *self.last_committed_leader_round_rx.borrow() {
-            if pos.block.round + CONSENSUS_STATUS_RETENTION_ROUNDS < last_committed_leader_round {
+            if pos.block.round + CONSENSUS_STATUS_RETENTION_ROUNDS <= last_committed_leader_round {
                 // Ignore stale status updates.
                 return;
             }
         }
 
+        let mut inner = self.inner.write();
+        self.set_transaction_status_locked(&mut inner, pos, status);
+    }
+
+    fn set_transaction_status_locked(
```

Comment:
nit: the `set_transaction_status_locked` gives me the impression that this method is trying to set the transaction status to "locked". A few suggestions:
* `set_transaction_status_inner`
* `set_transaction_status_with_lock`